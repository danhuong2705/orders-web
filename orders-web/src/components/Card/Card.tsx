import React from "react";
import styled from "@emotion/styled";
import { IProduct } from "../../interface/Product";
import { formatCurrency } from "../../helpers";
interface CardProps {
  sku: string;
  name: string;
  price: number;
  imageUrl: string;
  onAddToCart: (product: IProduct) => void;
}
const Wrapper = styled.div`
  display: flex;
  flex-direction: column;
  margin: 24px 4px;
  align-items: center;
  justify-content: center;
  cursor: pointer;
`;
const Name = styled.div`
  font-weight: bold;
  letter-spacing: 1px;
  font-size: 14px;
  margin-top: 8px;
  text-align: center;
`;
const Image = styled.img`
  width: 315px;
  height: 425px;
  object-fit: contain;
  :hover {
    box-shadow: 1px 1px 7px 2px #e0e0e0;
  }
`;
const Price = styled.div`
  margin-top: 8px;
  font-size: 14px;
`;
const AddToCartBtn = styled.button`
  font-size: 2rem;
  background-color: #fff;
  width: 100%;
  margin-top: 1rem;
  border: 1px solid #e0e0e0;
  cursor: pointer;
  :hover {
    border: 1px solid #000000;
  }
`;
const Card: React.FC<CardProps> = ({
  sku,
  name,
  price,
  imageUrl,
  onAddToCart,
}: CardProps) => {
  return (
    <Wrapper>
      <Image src={imageUrl} />
      <Name>{name}</Name>
      <Price>
        {formatCurrency(price)}
      </Price>
      <AddToCartBtn onClick={() => onAddToCart({ sku, price, imageUrl, name })}>
        +
      </AddToCartBtn>
    </Wrapper>
  );
};

export default Card;
