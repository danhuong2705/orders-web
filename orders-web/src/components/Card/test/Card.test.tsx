import React from "react";
import { cleanup, render } from "@testing-library/react";
import { BrowserRouter } from "react-router-dom";
import Card from "..";

afterEach(() => {
  jest.clearAllMocks();
  cleanup();
});

describe("<Card />", () => {
  it("should render correctly", async () => {
    const { container } = render(
      <BrowserRouter>
        <Card
          sku="21232"
          name="121323"
          imageUrl=""
          price={20000}
          onAddToCart={() => {}}
        />
      </BrowserRouter>
    );
    expect(container.firstChild).toMatchSnapshot();
  });
});
