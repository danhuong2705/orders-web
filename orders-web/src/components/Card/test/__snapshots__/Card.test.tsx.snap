// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Card /> should render correctly 1`] = `
<div
  class="css-f6s1up"
>
  <img
    class="css-1i5qv4l"
    src=""
  />
  <div
    class="css-4db4lp"
  >
    121323
  </div>
  <div
    class="css-16w20my"
  >
    20.000 ₫
  </div>
  <button
    class="css-hwo0d0"
  >
    +
  </button>
</div>
`;
